### PR TITLE
[zero] hybrid cpu adam

### PR DESCRIPTION
**Pull request by feifeibear**
Make our customized CPUADAM more general. It did not run it on CUDA. Now it is possible.

### CODE REVIEW

File: colossalai/nn/optimizer/cpu_adam.py [hunk 1]
```diff
@@ -31,18 +31,46 @@ def __init__(self,
...
+                          beta1,
+                          beta2,
+                          eps,
+                          weight_decay,
+                          bias_correction1,
+                          bias_correction2,
+                          loss_scale,
+                          use_adamw=False):
+        if loss_scale is not None:
+            # grad.div_(loss_scaler.loss_scale)
```
Comment (FrankLeeeee) [inline colossalai/nn/optimizer/cpu_adam.py:54]: remove this line

File: colossalai/nn/optimizer/cpu_adam.py [hunk 2]
```diff
@@ -31,18 +31,46 @@ def __init__(self,
...
-                                     eps,
-                                     weight_decay,
-                                     adamw_mode,
-                                     simd_log)
+        self.cpu_adam_op.create_adam(self.opt_id, lr, betas[0], betas[1], eps, weight_decay, adamw_mode, simd_log)
 
     def __del__(self):
         self.cpu_adam_op.destroy_adam(self.opt_id)
 
+    def torch_adam_update(self,
```
Comment (ver217) [inline colossalai/nn/optimizer/cpu_adam.py:39]: Can we just use original `torch.optim.adam.adam` ?

Comment (feifeibear) [reply colossalai/nn/optimizer/cpu_adam.py:39]: I try to import the function, it failed
```
from torch.optim._functional.adam
```
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'torch.optim._functional.adam'; 'torch.optim._functional' is not a package
https://github.com/pytorch/pytorch/blob/0fc6bd2e47/torch/optim/_functional.py#L54

Comment (ver217) [reply colossalai/nn/optimizer/cpu_adam.py:39]: OK. I see.

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**